torch_xla/csrc/ops/index_op.cpp

```diff
@@ -0,0 +1,163 @@
+#include "torch_xla/csrc/ops/index_op.h"
+#include <ATen/ExpandUtils.h>
+#include <ATen/Functions.h>
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "torch_xla/csrc/aten_xla_bridge.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/arithmetic_ir_ops.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/ops/ops.h"
+#include "torch_xla/csrc/xla_lower_util.h"
+
+namespace torch_xla {
+namespace {
+
+void CheckIndexTensorTypes(at::TensorList indices) {
+  for (auto& tensor : indices) {
+    if (tensor.defined()) {
+      auto& type = tensor.type();
+      auto scalar_type = type.scalarType();
```

Comment:
This is not a template, is it? So maybe at::ScalarType ...

Reply:
Done.